apps/prairielearn/elements/pl-file-editor/pl-file-editor.js

```diff
@@ -28,7 +28,14 @@ window.PLFileEditor = function (uuid, options) {
   this.editor.getSession().on('change', this.syncFileToHiddenInput.bind(this));
 
   if (options.aceMode) {
+    const oldModePath = ace.config.get('modePath');
```

Comment:
I don't quite understand this code. Why can't we just do `ace.config.set('modePath', options.aceModePath);` unconditionally?

Reply:
> Maybe call this `globalModePath` instead of `oldModePath`? That would clarify what is going on for me.

What about `defaultModePath`?

Reply:
I actually found a cleaner way to do it. I can specify the URL to be used for a particular mode by using [`ace.config.setModuleUrl()`](https://github.com/ajaxorg/ace/blob/35e1be52fd8172405cf0f219bab1ef7571b3363f/src/config.js#L95). This way I don't need to mess up with the global config settings and the path will only apply to the current mode. Done in 470afdae7c7e0a136e2a4f4f4f447bf7662b6f13.